app/src/androidTest/java/com/epfl/drawyourpath/mainpage/fragments/ChallengeFragmentTest.kt

```diff
@@ -148,312 +191,159 @@ fun modifyPathGoalWithBlankDoesNothing() {
-    val scenario = FragmentScenario.launchInContainer(
-        ChallengeFragment::class.java,
-        Bundle(),
-        R.style.Theme_Bootcamp,
-    )
-
-    // change value of time
-    onView(withId(R.id.goals_view))
-        .perform(
+        val newTime = 70
+
+        // change value of path
+        onView(withId(R.id.goals_view)).perform(
             actionOnItemAtPosition<RecyclerView.ViewHolder>(
                 1,
-                replaceTextOnViewChild("", R.id.goal_display_edit_text),
+                replaceTextOnViewChild(newTime.toString(), R.id.goal_display_edit_text),
             ),
             actionOnItemAtPosition<RecyclerView.ViewHolder>(
                 1,
                 pressImeActionButtonOnViewChild(R.id.goal_display_edit_text),
             ),
         )
 
-    val timeProgressGoal = context.resources.getString(R.string.progress_over_goal).format(0.0, mockUser.goals!!.activityTime!!)
+        val pathProgressGoal = context.resources.getString(R.string.progress_over_goal).format(0.0, newTime.toDouble())
 
-    waitUntilAllThreadAreDone()
+        waitUntilAllThreadAreDone()
 
-    // check that the value is correctly changed
-    onView(withId(R.id.goals_view))
-        .perform(scrollToPosition<RecyclerView.ViewHolder>(1))
-        .check(matches(hasDescendant(withText(timeProgressGoal))))
-        .check(matches(hasDescendant(withText(mockUser.goals!!.activityTime!!.toInt().toString()))))
-        .check(matches(hasDescendant(withText(R.string.minutes))))
+        // check that the value is correctly changed
+        onView(withId(R.id.goals_view)).check(matches(hasDescendant(withText(pathProgressGoal))))
+            .check(matches(hasDescendant(withText(newTime.toString()))))
+            .check(matches(hasDescendant(withText(R.string.minutes))))
 
-    scenario.close()
-}
+        scenario.close()
+    }
 
-/**
- * test if the time goal is correctly displayed when there is today daily goal
- */
-@Test
-fun displayTimeGoalOfTodayDailyGoal() {
-    val bundle = Bundle()
-    bundle.putBoolean(ChallengeFragment.TEST_EXTRA, true)
-
-    val scenario = FragmentScenario.launchInContainer(
-        ChallengeFragment::class.java,
-        bundle,
-        R.style.Theme_Bootcamp,
-    )
-
-    val timeProgressGoal =
-        context.resources.getString(R.string.progress_over_goal).format(dailyGoal.time, dailyGoal.expectedTime)
-
-    waitUntilAllThreadAreDone()
-
-    // check that time is correct
-    onView(withId(R.id.goals_view))
-        .perform(scrollToPosition<RecyclerView.ViewHolder>(1))
-        .check(matches(hasDescendant(withText(timeProgressGoal))))
-        .check(matches(hasDescendant(withText(dailyGoal.expectedTime.toInt().toString()))))
-        .check(matches(hasDescendant(withText(R.string.minutes))))
-
-    scenario.close()
-}
+    @Test
+    fun modifyPathsGoalDisplayNewPathsGoal() {
+        val scenario = FragmentScenario.launchInContainer(
+            ChallengeFragment::class.java,

```

Comment:
Same here : This test is commented. Maybe add a TODO to explain the reason or delete it.